lib/ansible/modules/network/nxos/nxos_lacp.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2019 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#############################################
+#                WARNING                    #
+#############################################
+#
+# This file is auto generated by the resource
+#   module builder playbook.
+#
+# Do not edit this file manually.
+#
+# Changes to this file will be over written
+#   by the resource module builder.
+#
+# Changes should be made in the model used to
+#   generate this file or in the resource module
+#   builder template.
+#
+#############################################
+
+"""
+The module file for nxos_lacp
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'network'
+                    }
+
+DOCUMENTATION = """
+---
+module: nxos_lacp
+version_added: 2.9
+short_description: Manage Global Link Aggregation Control Protocol (LACP) on Cisco NX-OS devices.
+description: This module manages Global Link Aggregation Control Protocol (LACP) on NX-OS devices.
+author: Trishna Guha (@trishnaguha)
+notes:
+  - Tested against NXOS 7.3.(0)D1(1) on VIRL.
+  - Feature lacp should be enabled for this module.
+options:
+  config:
+    description: LACP global options.
+    type: dict
+    suboptions:
+      system:
+        description:
+          - LACP system options
+        type: dict
+        suboptions:
+          priority:
+            description:
+              - The system priority to use in LACP negotiations.
+            type: int
```

Comment:
here `priority`/`mac` should be `required` param, correct?

Reply:
@justjais No there is no required parameter here. If no parameter is passed, the module won't do anything, it is not required or mandatory for nxos lacp. lacp can still be configured for interfaces if it's enabled even if these parameters are not configured.